src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -232,7 +232,11 @@ private static void RemoveIntersectingLocals(Dictionary<LocalSymbol, LocalDefUse
     internal class LocalDefUseInfo
     {
         // stack at variable declaration, may be > 0 in sequences.
-        public readonly int stackAtDeclaration;
+        public int StackAtDeclaration { get; private set; }
+        private readonly ObjectPool<LocalDefUseInfo> _pool;
```

Comment:
That might be inconvenient since they are mutable. 
Although while implementing this change I noticed that we do most mutations in just one place, so it might not be a big deal.
I will think about this.